src/repair/functions.js

```diff
@@ -46,11 +45,21 @@ export function repairFunctions() {
       throw e;
     }
     const FunctionPrototype = getPrototypeOf(FunctionInstance);
+    const oldFunctionConstructor = FunctionPrototype.constructor;
 
+    function isRunningInRealms() {
+      const e = new Error().stack;
+      if (!e) return true;
+      return e.indexOf('eval') !== -1;
+    }
     // Prevents the evaluation of source when calling constructor on the
     // prototype of functions.
     const TamedFunction = function() {
-      throw new TypeError('Not available');
+      if (isRunningInRealms()) {
```

Comment:
Switching anything semantically significant on the caller is hugely hazardous, and likely is a form of dynamic scoping, which we have taken pains to avoid in the rest of the language. Also, the `stack` property of Errors is not yet standard, and is likely to change.

Please explain what concrete compatibility problems you are running into, and what you need them to do. Let's see how we can accommodate those cases, or how you can on top of the realms-shim, with minimal damage to the realms-shim itself. Thanks.